### PR TITLE
chore(models): unify mocks setup, improve naming, test coverage and quality

### PR DESCRIPTION
**Pull request by Tlacenka**
This is the first PR which aims to improve the quality of the models package defined in #254 . In it I do the following:
- Add fs mock setup to all remaining unit tests.
- Remove generated testing-utils unit tests.
- Base models file naming on glossary.
- Rename `AuditGroup` to `Group` for all related entities.
- Clean up tests for smaller models (issue, audit, group, persistConfig, uploadConfig).

### CODE REVIEW

File: tsconfig.base.json [hunk 1]
```diff
@@ -25,9 +25,6 @@
       "@code-pushup/lighthouse-plugin": [
         "packages/plugin-lighthouse/src/index.ts"
       ],
-      "@code-pushup/lighthouse-plugin/testing": [
-        "packages/plugin-lighthouse/test/index.ts"
-      ],
       "@code-pushup/models": ["packages/models/src/index.ts"],
       "@code-pushup/models/testing": ["packages/models/test/index.ts"],
```
Comment (matejchalk) [inline tsconfig.base.json:29]: This one should also be deleted, I believe:

```suggestion
```

**Review by matejchalk — DISMISSED** [review]
I :heart: this refactor! Perfect example of how keeping things simple does wonders for readability :ok_hand: ([DAMP not DRY](https://stackoverflow.com/a/11837973/4607488)!)

File: tsconfig.base.json (hunk 1 above)
Comment (Tlacenka) [reply tsconfig.base.json:29]: Not yet, this PR is part 1 of 2. I still need to clean up all the main config types and move `test` folder to `testing-utils`.

File: packages/models/src/lib/implementation/utils.unit.test.ts [hunk 2]
```diff
@@ -58,27 +54,34 @@ describe('filenameRegex', () => {
...
     'file|name',
-  ])(`should not match invalid file name %p`, invalidFilename => {
+  ])(`should not match an invalid file name %p`, invalidFilename => {
     expect(invalidFilename).not.toMatch(filenameRegex);
   });
 });
 
-describe('stringUnique', () => {
+describe('hasDuplicateStrings', () => {
   it('should return true for a list of unique strings', () => {
```
[the remark this answers is not in the document]
Comment (Tlacenka) [reply packages/models/src/lib/implementation/utils.unit.test.ts:63]: Oops 😅

**Review by matejchalk — APPROVED** [review]

**Pull request merged**